### PR TITLE
Fix(eos_designs): Improve error message for missing device facts

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Improve error message for missing device facts

## Component(s) name

`arista.avd.eos_designs`

## How to test

Implemented negative test to exercise this error.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
@kmueller68

**Comment by ClausHolbechArista** [comment]
@kmueller68

**Review by gmuloc — APPROVED** [review]

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/utils.py [hunk 1]
```diff
@@ -34,7 +34,10 @@ def get_peer_facts(self: SharedUtils, peer_name: str, required: bool = True) ->
             f"avd_switch_facts..{peer_name}..switch",
             separator="..",
             required=required,
-            org_key=f"avd_switch_facts.{peer_name}.switch",
+            org_key=(
+                f"Facts not found for node '{peer_name}'. Something in the input vars is pointing to this node. "
+                f"Check that '{peer_name}' is in the inventory and is part of the group set by 'fabric_name'. Node"
```
Comment (pvinci-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/utils.py:39]: Would it not be clearer to the user to return the actual `mlag.domain_id`. ("domain_id": self.shared_utils.group) name?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/utils.py:39]: This function is shared between all "peerings" where we need to look up the information on the peer device. So we cannot do anything specific to the case. We got a report from a customer where it was route-servers triggering this issue. I just found it simpler to test with mlag.

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**